src/main/java/task/ListOfTask.java

```diff
@@ -0,0 +1,171 @@
+        listOfTask.add(temp);
+        if (print) {
+            System.out.println("added: " + temp);
+            Storage.save(listOfTask);
+        }
+    }
+
+    /**
+     * Prints out the list of task.
+     */
+    public void listTasks() {
+        int[] i = new int[1];
+        i[0] = 1;
+        listOfTask.forEach(x -> {
+            System.out.print(i[0] + ".");
+            System.out.println(x);
+            i[0]++;
+        });
+    }
+
+
+    /**
+     * Finds and prints index and tasks that contains the string str in its name.
+     *
+     * @param str The string that will be searched.
+     */
+    public void find(String str) {
+        int[] start = new int[1];
+        start[0] = 0;
+        int size = listOfTask.size();
+        int[] rememberIndex = new int[size];
+        Task[] rememberTask = new Task[size];
+        int[] i = new int[1];
+        i[0] = 1;
+        listOfTask.forEach(x -> {
+            if (x.getTaskName().contains(str)) {
+                rememberTask[start[0]] = x;
+                rememberIndex[start[0]] = i[0];
+                start[0]++;
+            }
+            i[0]++;
+        });
+        for (int j = 0; j < (start[0]); j++) {
+            System.out.println(rememberIndex[j] + "." + rememberTask[j]);
+        }
+        if (start[0] == 0) {
+            System.out.println("Whoopys uWu, sorry I couldnyt fynd any taysk that contyain that strying. XD uWu");
+        }
+    }
+
+    /**
+     * Mark a task as done.
+     *
+     * @param index The index of the task based on the current task list.
+     * @param print True to print messages, false to not print messages.
+     * @throws DukeException If the number is outside the range of indexes in the list.
+     */
+    public void mark(int index, boolean print) throws DukeException {
+        try {
+            listOfTask.get(index - 1).setDone();
+            if (print) {
+                System.out.println(listOfTask.get(index - 1).toString());
+                Storage.save(listOfTask);
+            }
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please select from index 1 to " + listOfTask.size());
+        }
+    }
+
+    /**
+     * Mark a task as undone.
+     *
+     * @param index The index of the task based on the current task list.
+     * @param print True to print messages, false to not print messages.
+     * @throws DukeException If the number is outside the range of indexes in the list.
+     */
+    public void unMark(int index, boolean print) throws DukeException {
+        try {
+            listOfTask.get(index - 1).setNotDone();
+            if (print) {
+                System.out.println(listOfTask.get(index - 1).toString());
+            }
+            Storage.save(listOfTask);
+        } catch (IndexOutOfBoundsException e) {
+            throw new DukeException("Please select from index 1 to " + listOfTask.size());
+        }
+    }

```

Comment:
To reduce some code duplication, you might consider merging both the `mark()` and `unMark()` methods and using an additional variable to decide whether to mark or unmark a particular task.

Reply:
Got it